docs/internal/GeneralArchitectureGuide.md

```diff
@@ -6,6 +6,59 @@
 
 ## Settings
 
+Elasticsearch supports [node-level configuration file settings][], [cluster-level settings][] and [index-level settings][].
+
+### Declaring a Setting
+
+[node-level configuration file settings]: https://www.elastic.co/guide/en/elasticsearch/reference/current/settings.html
+[cluster-level settings]: https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-update-settings.html
+[index-level settings]: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-update-settings.html
+
+The [Setting][] class is the building block for Elasticsearch server settings. Each `Setting` can take multiple [Property][]
+declarations to define setting characteristics, like allowing [dynamic updates][] while the node is running, or the scope of
+the setting to apply at the individual [index-level][] vs [cluster-level][]. A [dynamic Setting][] is often paired with a
+[local volatile variable][] and [ClusterSettings#initializeAndWatch()][] to catch and immediately apply dynamic changes.
+
+[Setting]: https://github.com/elastic/elasticsearch/blob/8.13/server/src/main/java/org/elasticsearch/common/settings/Setting.java#L57-L80
```

Comment:
I'm not sure if we need to tag specific code version in the doc. Few releases after this doc will still be valid but it will point on kinda obsolete code.

Reply:
Oops, I need to use a stable point like v8.13.2 (not 8.13 branch that can get backports) so the lines won't move to point at different code later. Updated.

Yeah, the links can become stale. But they will match the documentation explanation (equal staleness if there are significant changes), and otherwise I cannot refer to particular lines of code. The high level explanation will still apply, even if the code shifts around in future.

Reply:
Link above references the javadoc of the org.elasticsearch.common.settings.Setting class.
I think it is enough to reference the class itself to be able to find corresponding javadocs without having to specify exact lines.

Reply:
I removed some links in my latest PR changes. I've become less link-happy than when I originally wrote this. But I'd like to keep this one for symmetry with the subsequent Property reference. I don't think it does any harm.

It got interesting later where I talk about `Metadata`: there are two classes with that name 😁